### PR TITLE
fix(hash): fix hash_key_data_size

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

as title

## Checklist

- ~[ ] I have written necessary rustdoc comments~
- ~[ ] I have added necessary unit tests and integration tests~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/common/src/hash/dispatcher.rs [hunk 1]
```diff
@@ -58,6 +58,34 @@ pub trait HashKeyDispatcher {
...
+pub fn hash_key_data_size(data_type: &DataType) -> DataSize {
+    use std::mem::size_of;
+
+    use crate::types::{
+        Decimal, IntervalUnit, NaiveDateTimeWrapper, NaiveDateWrapper, NaiveTimeWrapper,
+        OrderedF32, OrderedF64,
+    };
+
+    match data_type {
+        DataType::Boolean => DataSize::FixedBits(1),
```
Comment (liurenjie1024) [inline src/common/src/hash/dispatcher.rs:70]: 1. We should also move `DataSize` and rename it to `HashKeySize`?
2. After 1, boolean should be `Fixed(1)`

Comment (Li0k) [reply src/common/src/hash/dispatcher.rs:70]: about 2
how can we distinguish between 1 bytes and  1 bit ?

Comment (Li0k) [reply src/common/src/hash/dispatcher.rs]: fix above

**Review by liurenjie1024 — APPROVED** [review]
LGTM

**Pull request merged**